go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -683,6 +683,34 @@ func (bnc *BaseNetworkController) isLocalZoneNode(node *kapi.Node) bool {
 	return util.GetNodeZone(node) == bnc.zone
 }
 
+// getActiveNetworkForNamespace returns the active network for the given namespace
+// and is a wrapper around util.GetActiveNetworkForNamespace
+func (bnc *BaseNetworkController) getActiveNetworkForNamespace(namespace string) (string, error) {
+	return util.GetActiveNetworkForNamespace(namespace, bnc.watchFactory.NADInformer().Lister())
+}
+
+// isPrimaryNetwork returns if pod's primary network is same
+// as this controller's network
+func (bnc *BaseNetworkController) isPrimaryNetwork(pod *kapi.Pod) (bool, error) {
+	if !util.IsNetworkSegmentationSupportEnabled() {
+		// if user defined network segmentation is not enabled
+		// then we know pod's primary network is "default" and
+		// pod's secondary network is not its NOT primary network
+		return !bnc.IsSecondary(), nil
+	}
+	activeNetwork, err := bnc.getActiveNetworkForNamespace(pod.Namespace)
+	if err != nil {
+		return false, err
+	}
+	if activeNetwork == types.UnknownNetworkName {
+		return false, fmt.Errorf("unable to determine what is the"+
+			"primary network for this pod %s; please remove multiple primary network"+
+			"NADs from namespace %s", pod.Name, pod.Namespace)
+		// TODO emit event
+	}
+	return activeNetwork == bnc.GetNetworkName(), nil
+}
```

Comment:
We should not duplicate this logic, it looks small but it's quite important, can we move it to a common place ? and use it at controller and pod allocator ?

Reply:
we need to add events recorder at pod allocator to be able to do this, opened a tech debt card to get back to this and spoke to @qinqon offline